Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4996,6 +4996,44 @@ void WebPageProxy::getAllFrames(CompletionHandler<void(FrameTreeNodeData&&)>&& c
     m_mainFrame->getFrameInfo(WTFMove(completionHandler));
 }
 
+void WebPageProxy::getAllFrameTrees(CompletionHandler<void(Vector<FrameTreeNodeData>&&)>&& completionHandler)
+{
+    class FrameTreeCallbackAggregator : public RefCounted<FrameTreeCallbackAggregator> {
+    public:
+        static Ref<FrameTreeCallbackAggregator> create(CompletionHandler<void(Vector<FrameTreeNodeData>&&)>&& completionHandler, size_t processCount) { return adoptRef(*new FrameTreeCallbackAggregator(WTFMove(completionHandler), processCount)); }
+        void addFrameTree(FrameTreeNodeData&& data) { m_data.uncheckedAppend(WTFMove(data)); }
```

Comment:
Can we make it private? It seems uncheckedAppend as a public method is not great.

Reply:
No, it needs to be public.  But I can make the append a checked append call.